packages/vm/src/cache.rs

```diff
@@ -185,6 +188,8 @@ where
         backend: Backend<A, S, Q>,
         options: InstanceOptions,
     ) -> VmResult<Instance<A, S, Q>> {
+        let _lock = self.m.lock().unwrap();
```

Comment:
Why unwrap here? Seems good for mvp, but are we sure we never panic? This could happen in the go callbacks for instance.

https://doc.rust-lang.org/std/sync/struct.Mutex.html#errors

Anyway, this is good to start. And maybe the best we can do.

I'd love to see results on wasmvm failure with this enabled.

Reply:
For what I've read, `unwrap()` is there to guarantee that if the thread holding the lock panics, the blocked threads panic too. If we're sure the mutex block never panics, we can remove the `unwrap()`, yes. Anyway, it doesn't hurt to have it, IMO, as the overhead is minimal.

With this enabled the wasmvm failure is gone, as far as I can tell. The only way to know for sure would be to analyze the complete concurrent code paths looking for races. But I've left the tests with this enabled running for hours, without issues.

Reply:
> I've left the tests with this enabled running for hours, without issues

Great news